go-controller/pkg/ovn/kubevirt.go

```diff
@@ -110,18 +122,287 @@ func (oc *DefaultNetworkController) deleteDHCPOptions(pod *kapi.Pod) error {
+	}
+	// Everything has already being cleand up since this is an old migration
+	// pod
+	if isMigratedSourcePodStale {
+		return nil
+	}
+	// This pod is not part of ip migration so we don't need to clean up
+	if !kubevirt.IsPodLiveMigratable(pod) {
+		return nil
+	}
+	if err := oc.deleteDHCPOptions(pod); err != nil {
+		return err
+	}
+	if err := oc.deleteRoutingForMigratedPod(pod); err != nil {
+		return err
+	}
+	return nil
+}
+
+// ensureLocalZonePodAddressesToNodeRoute will add static routes and policies to ovn_cluster_route logical router
+// to ensure VM traffic work as expected after live migration if the pod is running at the local/global zone
+// following is the list of NB logical resources created depending if it's interconnected or not:
+// IC:
+//   - static route with VM ip as dst-ip prefix and output port as node switch LRP
+//   - static route with cluster wide CIDR as src-ip prefix and nexthop GR, it has less
+//     priority than route to use overlay in case of pod to pod communication
+//
+// NO IC:
+//   - static route with VM ip as dst-ip prefix and output port as node switch LRP
+//   - low priority policy with src VM ip and reroute GR, since it has low priority
+//     it will not override the policy to enroute pod to pod traffic using overlay
+func (oc *DefaultNetworkController) ensureLocalZonePodAddressesToNodeRoute(pod *kapi.Pod) error {
+	podAnnotation, err := util.UnmarshalPodAnnotation(pod.Annotations, types.DefaultNetworkName)
+	if err != nil {
+		return fmt.Errorf("failed reading ovn annotation: %v", err)
+	}
+	switchNames, err := oc.getSwitchNames(pod)
+	if err != nil {
+		return fmt.Errorf("failed configuring pod routing when getting switch current and original name: %v", err)
+	}
+
+	// VM is running at the node that owns the subnet the point to point
+	// routing is not needed
+	if switchNames.Current == switchNames.Original {
+		// Delete not needed policies or static routes
+		if err := libovsdbops.DeleteLogicalRouterPoliciesWithPredicate(oc.nbClient, types.OVNClusterRouter, func(policy *nbdb.LogicalRouterPolicy) bool {
+			policyVMLabel, ok := policy.ExternalIDs[string(libovsdbops.VirtualMachineIndex)]
+			return ok && policyVMLabel == pod.Labels[kubevirtv1.VirtualMachineNameLabel]
+		}); err != nil {
+			return fmt.Errorf("failed configuring pod routing when deleting stale policies for pod %s/%s: %v", pod.Namespace, pod.Name, err)
+		}
+		if err := libovsdbops.DeleteLogicalRouterStaticRoutesWithPredicate(oc.nbClient, types.OVNClusterRouter, func(route *nbdb.LogicalRouterStaticRoute) bool {
+			routeVMLabel, ok := route.ExternalIDs[string(libovsdbops.VirtualMachineIndex)]
+			return ok && routeVMLabel == pod.Labels[kubevirtv1.VirtualMachineNameLabel]
+		}); err != nil {
+			return fmt.Errorf("failed configuring pod routing when deleting stale static routes for pod %s/%s: %v", pod.Namespace, pod.Name, err)
+		}
+		return nil
```

Comment:
Why cant we reuse deleteRoutingForMigratedPod somehow?

Reply:
We can, I will do it.